### PR TITLE
storage: cleanup traits for v0.2

### PR DESCRIPTION
**Pull request by huachaohuang**
Check #143 for example usage of Storage.

### CODE REVIEW

File: src/storage/src/mem/bucket.rs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+}
+
+pub struct SequentialReader {
+    object: Object,
+    offset: usize,
+}
+
+impl SequentialReader {
+    fn new(object: Object) -> Self {
+        Self { object, offset: 0 }
```
Comment (zojw) [inline src/storage/src/mem/bucket.rs:81]: maybe we still need export offset as parameter in furture? (just a question not review

**Review by zojw — COMMENTED** [review]
LGTM, I'd like to fix file impl tomorrow

**Review by zojw — APPROVED** [review]
LGTM, I'd like to fix file impl tomorrow

File: src/storage/src/mem/bucket.rs (hunk 1 above)
Comment (huachaohuang) [reply src/storage/src/mem/bucket.rs:81]: Yes, maybe. Or a random reader.

**Pull request merged**